### PR TITLE
updated test cases to use MultithreadTestCase

### LINKED ISSUE #108744: switch more test cases to use MultithreadTestCase
MultithreadTestCase allow us to run less resouce by spawning threads instead of processes, which could make distributed tests run faster. We have the following test files still not using MultithreadTestCase, and we should switch those test case to use it.

[ ] https://github.com/pytorch/pytorch/blob/main/test/distributed/_tensor/test_math_ops.py
[ ] https://github.com/pytorch/pytorch/blob/main/test/distributed/_tensor/test_matrix_ops.py
[ ] https://github.com/pytorch/pytorch/blob/main/test/distributed/_tensor/test_tensor_ops.py
[ ] https://github.com/pytorch/pytorch/blob/main/test/distributed/_tensor/test_embedding_ops.py

Example test case that already uses multithreaded test case, see https://github.com/pytorch/pytorch/blob/main/test/distributed/_tensor/test_pointwise_ops.py#L75

one just need to extend the `DTensorOpTestBase` for the above test files, should be relatively simple

### PR DESCRIPTION
**Pull request by SaumyaBhushan**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=VSV8ekm)
Fixes #108744

### CODE REVIEW

**Comment by SaumyaBhushan** [comment]
Hi @zou3519 @rohan-varma @H-Huang Please review this PR

File: test/distributed/_tensor/test_matrix_ops.py [hunk 1]
```diff
@@ -15,13 +15,13 @@
...
     skip_unless_torch_gpu,
     with_comms,
 )
 
 
-class DistMatrixOpsTest(DTensorTestBase):
+class DistMatrixOpsTest(DTensorOpTestBase):
     @with_comms
     def test_addmm(self):
         device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
```
Comment (wanchaol) [inline test/distributed/_tensor/test_matrix_ops.py:27]: can we also change the `device_mesh` init to use `self.build_mesh`, thanks!

File: test/distributed/_tensor/test_tensor_ops.py [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
-    DTensorTestBase,
+    DTensorOpTestBase,
     with_comms,
 )
 
 
-class DistTensorOpsTest(DTensorTestBase):
+class DistTensorOpsTest(DTensorOpTestBase):
     @with_comms
     def test_aten_contiguous(self):
```
Comment (wanchaol) [inline test/distributed/_tensor/test_tensor_ops.py:17]: same here, let's use build_mesh instead.

File: test/distributed/_tensor/test_math_ops.py [hunk 3]
```diff
@@ -9,13 +9,13 @@
...
-    DTensorTestBase,
+    DTensorOpTestBase,
     skip_unless_torch_gpu,
     with_comms,
 )
 
 
-class DistMathOpsTest(DTensorTestBase):
+class DistMathOpsTest(DTensorOpTestBase):
     @with_comms
```
Comment (wanchaol) [inline test/distributed/_tensor/test_math_ops.py:19]: I don't think we need the `with_comms` wrapper when switching to OpTestBase, please fix those

File: test/distributed/_tensor/test_matrix_ops.py [hunk 4]
```diff
@@ -15,13 +15,13 @@
...
-    DTensorTestBase,
+    DTensorOpTestBase,
     skip_unless_torch_gpu,
     with_comms,
 )
 
 
-class DistMatrixOpsTest(DTensorTestBase):
+class DistMatrixOpsTest(DTensorOpTestBase):
     @with_comms
```
Comment (wanchaol) [inline test/distributed/_tensor/test_matrix_ops.py:25]: same, can remove `with_comms`

File: test/distributed/_tensor/test_tensor_ops.py [hunk 5]
```diff
@@ -7,12 +7,12 @@
...
     DTensorConverter,
-    DTensorTestBase,
+    DTensorOpTestBase,
     with_comms,
 )
 
 
-class DistTensorOpsTest(DTensorTestBase):
+class DistTensorOpsTest(DTensorOpTestBase):
     @with_comms
```
Comment (wanchaol) [inline test/distributed/_tensor/test_tensor_ops.py:16]: ditto

**Review by wanchaol — CHANGES_REQUESTED** [review]
Thanks for contributing! a few comments inlined.

**Pull request closed without merging**